app/controllers/api/base_controller/authentication.rb

```diff
@@ -52,6 +52,8 @@ def authorize_user_group(user_obj)
           group_obj = user_obj.miq_groups.find_by(:description => group_name)
           raise AuthenticationError, "Invalid Authorization Group #{group_name} specified" if group_obj.nil?
           user_obj.current_group_by_description = group_name
+        elsif user_obj.current_group.nil? && user_obj.miq_groups.present?
```

Comment:
I have a couple of concerns here: 

1. I don't think this restores the previous behavior, it seems like a workaround to prevent it from blowing up. It is sort of a new feature that is fixing a bug, and I'd prefer those two things to be separate PRs
2. I have a larger concern that extends to other things in this method, which is that the API seems to know too much about authorization-domain things. I'd prefer for us just to pass the group name to some more central authorization code, which would handle the correcting of the user's current group if that's what we're deciding is what it should do, and should just return success/failure in some way.